sklearn/naive_bayes.py

```diff
@@ -958,3 +998,256 @@ def _joint_log_likelihood(self, X):
+
+            Must be provided at the first call to partial_fit, can be omitted
+            in subsequent calls.
+
+        Returns
+        -------
+        self : object
+        """
+        return super(CategoricalNB, self).partial_fit(X, y, classes,
+                                                      sample_weight=None)
+
+    def _check_input(self, X, Y=None):
+        if Y is None:
+            if _old_numpy:
+                X = check_array(X, accept_sparse=False)
+                if not np.isfinite(X).all():
+                    raise ValueError("Input contains NaN or infinity.")
+                return np.asarray(X, dtype=np.int64)
+            else:
+                return check_array(X, accept_sparse=False)
+        else:
+            if _old_numpy:
+                X, Y = check_X_y(X, Y, accept_sparse=False)
+                if not np.isfinite(X).all():
+                    raise ValueError("Input contains NaN or infinity.")
+                X = np.asarray(X, dtype=np.int64)
+                return X, Y
+            else:
+                return check_X_y(X, Y, accept_sparse=False)
+
+    def _init_counters(self, n_effective_classes, n_features):
+        self.class_count_ = np.zeros(n_effective_classes, dtype=np.float64)
+        self.category_count_ = [np.zeros((self.class_count_.shape[0], 0))
+                                for _ in range(n_features)]
+        self.feature_cat_index_mapping_ = [{} for _ in range(n_features)]
+
+    def _count(self, X, Y):
+        if np.any((X.data if issparse(X) else X) < 0):
+            raise ValueError("Input X must be non-negative")
+        self.class_count_ += Y.sum(axis=0)
+        for i in range(self.n_features_):
+            X_feature = X[:, i]
+            categories = np.unique(X_feature)
+            self._update_mapping(self.feature_cat_index_mapping_[i],
+                                 categories)
+            # update category_count_dimensions in case partial_fit is used
+            self.category_count_[i] = self._update_category_count_dimensions(
+                self.category_count_[i], self.feature_cat_index_mapping_[i])
+            self._update_category_count(X_feature, Y,
+                                        self.feature_cat_index_mapping_[i],
+                                        self.category_count_[i])
+
+    def _count_categories(self, X_feature):
+        if _old_numpy:
+            bins = np.bincount(X_feature)
+            non_zero_bins = np.nonzero(bins)[0]
+            return non_zero_bins, bins[non_zero_bins]
+        else:
+            return np.unique(X_feature, return_counts=True)
+
+    def _update_mapping(self, cat_mapping, categories):
+        for category in categories:
+            category = category
+            if category not in cat_mapping:
+                cat_mapping[category] = len(cat_mapping)
+
+    def _update_category_count_dimensions(self, cat_count, cat_mapping):
```

Comment:
Okay

Reply:
I would also change the order of parameters here to `cat_mapping, cat_count` since it is then more consistent to how you use it in other methods like `_update_category_count`.

Reply:
I would also change the order of parameters here to `cat_mapping, cat_count` since it is then more consistent to how you use it in other methods like `_update_category_count`.

Reply:
I would also change the order of parameters here to cat_mapping, cat_count since it is then more consistent to how you use it in other methods like _update_category_count.